### PR TITLE
Update analyzer versions per warnings in a build

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

**Review by brycewang-microsoft — APPROVED** [review]

File: vsts/vsts.yaml [hunk 1]
```diff
@@ -554,15 +554,15 @@ jobs:
...
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-policheck.PoliCheck@2
         displayName: "Run PoliCheck"
         inputs:
           targetType: F
           optionsRulesDBPath: 'vsts/PolicheckExclusionsDB.mdb'
           optionsSEV: '1|2|3|4'
           optionsPE: 1
       
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-antimalware.AntiMalware@3
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-antimalware.AntiMalware@4
```
Comment (drwill-ms) [inline vsts/vsts.yaml:565]: ##[warning]* Please upgrade the version of this build task in your build to major version: 4. For more information, please see: https://aka.ms/gdn-v1-deprecation

File: vsts/vsts.yaml [hunk 2]
```diff
@@ -554,15 +554,15 @@ jobs:
       - task: ComponentGovernanceComponentDetection@0
         displayName: "Component Detection"
       
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-policheck.PoliCheck@1
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-policheck.PoliCheck@2
```
Comment (drwill-ms) [inline vsts/vsts.yaml:557]: ##[warning]* Please upgrade the version of this build task in your build to major version: 2. For more information, please see: https://aka.ms/gdn-v1-deprecation

File: vsts/vsts.yaml [hunk 3]
```diff
@@ -571,13 +571,6 @@ jobs:
           # The support team already has a process to address this, so our pipeline can treat stale signatures as warnings (instead of treating it as an error).
           TreatStaleSignatureAs: Warning
 
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-autoapplicability.AutoApplicability@1
```
Comment (drwill-ms) [inline vsts/vsts.yaml:574]: ##[warning]* This tool is no longer supported. Please remove this build task from your build definition. For more information, please see: https://aka.ms/gdn-v1-deprecation

File: vsts/vsts.yaml [hunk 4]
```diff
@@ -594,7 +587,7 @@ jobs:
         # TODO #181 Config issue: must run on Debug builds only with valid PDBs.
         enabled: false
 
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@2
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-credscan.CredScan@3
```
Comment (drwill-ms) [inline vsts/vsts.yaml:590]: ##[warning]* Please upgrade the version of this build task in your build to major version: 3. For more information, please see: https://aka.ms/gdn-v1-deprecation

File: vsts/vsts.yaml [hunk 5]
```diff
@@ -604,13 +597,10 @@ jobs:
           # TODO #181 Samples / tests fail the test due to fake connection strings.
           debugMode: false
 
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-vulnerabilityassessment.VulnerabilityAssessment@0
```
Comment (drwill-ms) [inline vsts/vsts.yaml:607]: ##[warning]* This tool is no longer supported. Please remove this build task from your build definition. For more information, please see: https://aka.ms/gdn-v1-deprecation

File: vsts/vsts.yaml [hunk 6]
```diff
@@ -604,13 +597,10 @@ jobs:
...
-        displayName: "Run Vulnerability Assessment"
-
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@2
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@3
```
Comment (drwill-ms) [inline vsts/vsts.yaml:600]: ##[warning]* Please upgrade the version of this build task in your build to major version: 3. For more information, please see: https://aka.ms/gdn-v1-deprecation

File: vsts/vsts.yaml [hunk 7]
```diff
@@ -604,13 +597,10 @@ jobs:
...
         displayName: "Publish Security Analysis Logs"
 
-      - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@1
+      - task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@2
```
Comment (drwill-ms) [inline vsts/vsts.yaml:603]: ##[warning]* Please upgrade the version of this build task in your build to major version: 2. For more information, please see: https://aka.ms/gdn-v1-deprecation

**Review by abhipsaMisra — APPROVED** [review]

**Review by patilsnr — APPROVED** [review]

**Pull request merged**